lib/ex_ami/time_server.exs

```diff
@@ -0,0 +1,144 @@
+defmodule TimeServer do
+  use GenServer
+
+  defstruct agents: %{},
+            results: %{},
+            values: [],
+            doubles: []
+
+  @name __MODULE__
+  @num_cores 2
+
+  def child_spec(args) do
+    %{
+      id: "#{@name}",
+      start: {__MODULE__, :start_link, args},
+      restart: :transient,
+      shutdown: 60,
+      type: :worker
+    }
+  end
+
+  @spec start_link(any) :: no_return
+  def start_link(_) do
+    GenServer.start_link(__MODULE__, [], name: @name)
+  end
+
+  def status() do
+    GenServer.call(@name, :status)
+  end
+
+  def reset() do
+    GenServer.cast(@name, :reset)
+  end
+
+  def display() do
+    GenServer.cast(@name, :display)
+  end
+
+  def compute() do
+    GenServer.call(@name, :compute, 20000)
```

Comment:
The 20000 is because when I tried to run a million values and other similar sizes I was getting a timeout on the compute function, so since this is a test module I just extended the timeout. Then I found a gross inefficiency, but this still allows the module to deal with more data.

No need to group since this is pretty much a one-shot test module.